### PR TITLE
Feature/user class refactor

### PR DESCRIPTION
**Pull request by theblockstalk**

### CODE REVIEW

File: src/sdk/controllers/UserAuthorization.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        const pinKey = await this.keyManager.checkKey({
+            level: KeyManagerLevel.PIN,
+            challenge: pin,
+        });
+
+        if (!pinKey) throwError('Pin is incorrect', SdkErrors.PinInvalid);
+        return true;
+    }
+
+    async saveFingerprint(): Promise<void> {
```
Comment (sadiabbasi) [inline src/sdk/controllers/UserAuthorization.ts:86]: so in future will we go back on to fingerprint and pin?

Comment (theblockstalk) [reply src/sdk/controllers/UserAuthorization.ts:86]: yes definitely

**Pull request merged**